contracts/JBveBanny.sol

```diff
@@ -284,46 +293,106 @@ contract JBveBanny is ERC721Votes, ERC721Enumerable, Ownable, ReentrancyGuard, J
+      // Calculate the updated time when this lock will end (in seconds).
+      uint256 _updatedLockedUntil = block.timestamp + _data.updatedDuration - _timeRemaining;
+
+      // The new lock must be greater than the current lock.
+      if (_lockedUntil > _updatedLockedUntil) revert INVALID_LOCK_EXTENSION();
+
+      // fetch the stored packed value.
+      uint256 packedValue = _count;
+      // _duration in the bits 152-199.
+      packedValue |= _data.updatedDuration << 152;
+      // _lockedUntil in the bits 200-247.
+      packedValue |= _updatedLockedUntil << 200;
+      // _useJbToken in bit 248.
+      if (_useJbToken) packedValue |= 1 << 248;
+      // _allowPublicExtension in bit 249.
+      if (_allowPublicExtension) packedValue |= 1 << 249;
+
+      _packedSpecs[_data.tokenId] = packedValue;
+
+      emit ExtendLock(_data.tokenId, _data.updatedDuration, _updatedLockedUntil, msg.sender);
+    }
+  }
 
-    // fetch the stored packed value.
-    uint256 packedValue = _count;
-    // _duration in the bits 152-199.
-    packedValue |= _updatedDuration << 152;
-    // _lockedUntil in the bits 200-247.
-    packedValue |= _updatedLockedUntil << 200;
-    // _useJbToken in bit 248.
-    if (_useJbToken) packedValue |= 1 << 248;
+  /**
+    @notice
+    Allows banny holders to set whether or not anyone in the public can extend their locked position.
 
-    _packedSpecs[_tokenId] = packedValue;
+    @dev
+    Only an owner account or a designated operator can extend the lock of its tokens.
 
-    emit ExtendLock(_tokenId, _updatedDuration, _updatedLockedUntil, msg.sender);
+    @param _allowPublicExtensionData An array of locks to extend.
+  */
+  function setAllowPublicExtensionFlag(
+    JBAllowPublicExtensionData[] calldata _allowPublicExtensionData
+  ) external nonReentrant {
+    for (uint256 _i; _i < _allowPublicExtensionData; _i++) {
+      // Get a reference to the extension being iterated.
+      JBAllowPublicExtensionData _data = _allowPublicExtensionData[_i];
+
+      // Get the specs for the token ID.
+      (uint256 _count, uint256 _duration, uint256 _lockedUntil, bool _useJbToken, ) = getSpecs(
+        _data.tokenId
+      );
+
+      // Check if the msg.sender is either the position owner or is an operator.
+      _requirePermission(
+        ownerOf(_data.tokenId),
+        projectId,
+        JBStakingOperations.SET_PUBLIC_EXTENSION_FLAG
+      );
+
+      // fetch the stored packed value.
+      uint256 packedValue = _count;
+      // _duration in the bits 152-199.
+      packedValue |= _duration << 152;
+      // _lockedUntil in the bits 200-247.
+      packedValue |= _lockedUntil << 200;
+      // _useJbToken in bit 248.
+      if (_useJbToken) packedValue |= 1 << 248;
+      // _allowPublicExtension in bit 249.
+      if (_data.allowPublicExtension) packedValue |= 1 << 249;
+
+      _packedSpecs[_data.tokenId] = packedValue;
+
+      emit SetAllowPublicExtension(_data.tokenId, _data.allowPublicExtension, msg.sender);
```

Comment:
I think we need a check here which checks `_data.allowPublicExtension` is always true will add that